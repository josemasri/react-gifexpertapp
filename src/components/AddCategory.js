import React, { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleKeyPress = (e) => {
    if (e.key === "Enter") {
      if (inputValue.trim() === "") {
        return;
      }
      console.log("Enter key Pressed");
      setCategories((categories) => [inputValue.trim(), ...categories]);
      setInputValue("");
    }
  };

  return (
    <>
      <input
        type="text"
        placeholder="Search a Gif..."
        value={inputValue}
        onChange={handleChange}
        onKeyPress={handleKeyPress}
      />
    </>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
