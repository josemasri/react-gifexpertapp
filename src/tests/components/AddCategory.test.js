import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("Pruebas en <AddCategory />", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola mundo";
    input.simulate("change", { target: { value } });
    const inputAfter = wrapper.find("input");
    expect(inputAfter.prop("value")).toBe(value);
  });

  test("No debe de cambiar la información con Enter", () => {
    const input = wrapper.find("input");
    input.simulate("keypress", { key: "Enter" });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("debe de llamar el setCategories y limpiar la caja de texto", () => {
    const wrapper = shallow(<AddCategory setCategories={setCategories} />);
    const input = wrapper.find("input");
    const value = "Hola mundo";
    // Agregar texto al input
    input.simulate("change", { target: { value } });
    const inputAfter = wrapper.find("input");
    // Probar si se agrego el texto
    expect(inputAfter.prop("value")).toBe(value);
    // Presionar enter
    inputAfter.simulate("keypress", { key: "Enter" });
    // Probar si se llamo la funcion
    // Diferentes maneras
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    const inputCleared = wrapper.find("input");
    // Probar que el input se limpió
    expect(inputCleared.prop("value")).toBe("");
  });
});
