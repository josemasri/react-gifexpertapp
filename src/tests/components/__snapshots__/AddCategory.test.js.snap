// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Search a Gif..."
    type="text"
    value=""
  />
</Fragment>
`;
